rotkehlchen/db/upgrades/v31_v32.py

```diff
@@ -86,6 +87,52 @@ def _add_new_tables(cursor: 'Cursor') -> None:
 );""")  # noqa: E501
 
 
+def _update_history_entries_from_kraken(cursor: 'Cursor') -> None:
+    """The logic for kraken was adding additional entries for trades when fee + kfee was
+    being used. This function makes the state of the database consistent with the upgraded
+    logic by:
+    - Removing extra row additions
+    - Updating the sequence index to avoid duplicates
+    - Make sure that no other event has duplicated sequence indexes
+    """
+    cursor.execute("""
+    DELETE FROM history_events where location="B" AND asset="KFEE" AND
+     type="trade" AND subtype=NULL;
+    """)
+    cursor.execute("""
+    UPDATE history_events SET sequence_index = sequence_index + 1 WHERE asset="KFEE"
+     AND location="B" AND type="trade" AND subtype="fee"
+    """)
+
+    cursor.execute("""
+    SELECT e.event_identifier, e.sequence_index, e.identifier from history_events e JOIN (SELECT event_identifier,
```

Comment:
Have you tested that if there is multiple duplicates such as:

```
eventid1, 1
eventid1, 1
eventid2, 3
eventid2, 3
```
they will all be returned?

Reply:
yes I have tested

```
# The trade
"KRAKEN-REMOTE-ID3",0
"KRAKEN-REMOTE-ID3",1
"KRAKEN-REMOTE-ID3",3
"KRAKEN-REMOTE-ID3",2
"KRAKEN-REMOTE-ID3",3

# Other event
"KRAKEN-REMOTE-ID4",0
"KRAKEN-REMOTE-ID4",1
"KRAKEN-REMOTE-ID4",1
"KRAKEN-REMOTE-ID4",2
"KRAKEN-REMOTE-ID4",2
```